src/scripts/activate-user.ts

```diff
@@ -0,0 +1,197 @@
+        `${context} To transition user to ${activeStatus} status, please follow through with the activation workflow.`
+      );
+    }
+
+    case okta.UserStatus.LOCKED_OUT: {
+      return TE.left(
+        `${context} To transition user to ${activeStatus} status, please use the unlock command.`
+      );
+    }
+    case okta.UserStatus.PASSWORD_EXPIRED: {
+      return TE.left(
+        `${context} To transition user to ${activeStatus} status, please instruct user to login with temporary password and follow the password reset process.`
+      );
+    }
+    case okta.UserStatus.RECOVERY: {
+      return TE.left(
+        `${context} To transition user to ${activeStatus} status, please follow through with the activation workflow or restart the workflow using the reactivate-user command.`
+      );
+    }
+    case okta.UserStatus.SUSPENDED: {
+      return TE.left(
+        `${context} To transition user to ${activeStatus} status, please use the unsuspend-user command.`
+      );
+    }
+    // STAGED or DEPROVISIONED user status
+    default: {
+      return TE.right({
+        ...user,
+        status: userStatus,
+      });
+    }
+  }
+};
+
+/**
+ * Prints out what would happen if we were to activate the user.
+ * @param user - the user to dry run the activation for.
+ * @returns a TaskEither that resolves to the user.
+ */
+const dryRunActivateUser = (
+  user: ActivatableUser
+): TE.TaskEither<string, User> => {
+  // eslint-disable-next-line functional/no-expression-statement
+  console.info(
+    `User current status is [${user.status}]. Will attempt to activate [${user.id}] [${user.email}].`
+  );
+  return TE.right(user);
+};
+
+/**
+ * Activates the user.
+ * @param service - the service to use to activate the user.
+ * @param user - the user to activate.
+ * @returns a TaskEither that resolves to the activated user.
+ */
+const activateUser = (
+  service: UserService,
+  user: ActivatableUser
+): TE.TaskEither<string, User> => {
+  // eslint-disable-next-line functional/no-expression-statement
+  console.info(`User current status is [${user.status}].`);
+  return pipe(
+    service.activateUser(user.id),
+    TE.tapIO((user) => Console.info(`Activated [${user.id}] [${user.email}].`))
+  );
+};
+
+/**
+ * Activates a user, only works if user currently has the status: staged or deprovisioned.
+ * @param service - the service to use to activate the user.
+ * @param userId - the id of the user to activate.
+ * @param dryRun - if true, will not actually activate the user, but will print out what would happen.
+ * @returns a TaskEither that resolves to the activated user.
+ */
+export const activateUserHandler = (
+  service: UserService,
+  userId: string,
+  dryRun: boolean
+): TE.TaskEither<string, User> =>
+  pipe(
+    validateUserExist(service, userId),
+    TE.chain((user) => validateUserStatusPriorToActivation(user)),
+    TE.chain((user) =>
+      dryRun ? dryRunActivateUser(user) : activateUser(service, user)
```

Comment:
**thought:** in the future, think about whether you could pass a function that goes from `ActivatableUser`and `UserService` to a `TaskEither<..., User>` so that you can lift the `dryRun` switch up one more level.

Reply:
how do you mean?

Reply:
```
export const activateUserHandler = (
  service: UserService,
  userId: string,
  commandHandler: (user: ActivatableUser) => TE.TaskEither<string, User>
): TE.TaskEither<string, User> =>
  pipe(
    validateUserExist(service, userId),
    TE.chain((user) => validateUserStatusPriorToActivation(user)),
    TE.chain(commandHandler)
  );
```

and used like this:

```
const { userId, dryRun } = args;
const commandHandler = dryRun ?
  dryRunActivateUser :
  activateUser(service);
const result = await activateUserHandler(service, userId, commandHandler)();
```

Reply:
I don't want this change as part of this PR though. This is for you to think about.

Reply:
Oh thats what you meant